pkg/secrets/strategy.go

```diff
@@ -64,9 +64,6 @@ type Source struct {
 }
 
 func (s Source) MarshalRaw() interface{} {
-	if s.Key == "" {
```

Comment:
It's not immediately clear to me why this was removed and I'm concerned it may change how things are serialized. Did something make you need to remove it?

Reply:
I changed it back